### DISCUSSION TITLE
Support HTTP3 on other Windows versions: Win7, Win10

### DISCUSSION THREAD

**Discussion by alexandrehtrb** [dotnet/runtime#61690] [General] [answered]
Hello!

I really liked the new HTTP3 feature on .NET 6 and I want to use it!
But it is not available for Windows 7 and Windows 10, are there any plans for support? This leaves many users left behind.
The MSQuic repository says that it relies on SChannel for Windows, will SChannel be brought to other Windows versions, besides 11?
I opened a discussion thread on MSQuic repository too: https://github.com/microsoft/msquic/discussions/2151

**Comment by huoyaoyuan** [comment, marked as answer]
Windows 7 has reached EOL and there will be definitely no new features.
For Windows 10, it's better to ask the OS team. But I'm afraid there won't be. The OS core version hasn't changed since 19041.

> I really liked the new HTTP3 feature on .NET 6 and I want to use it!
But it is not available for Windows 7 and Windows 10, are there any plans for support? This leaves many users left behind.

For TLS and HTTP protocol, the best practice is to support a range of versions, and let the negotiation protocol to select the best version. This will support the largest range of users.

**Reply by alexandrehtrb** [reply]
> For TLS and HTTP protocol, the best practice is to support a range of versions, and let the negotiation protocol to select the best version. This will support the largest range of users.

Yes, we can offer HTTP/1.1 and HTTP/2 for users on those platforms, but they will not be able to use HTTP/3 on .NET.
More than 80% of Windows users are using Win10 and leave .NET applications out of HTTP/3 is bad...
This is also an advantage that other languages will have comparing to .NET. I use Firefox on Windows 7 and I can make HTTP/3 requests with it.

**Reply by karelz** [reply]
Also answered here: https://github.com/microsoft/msquic/discussions/2151#discussioncomment-1720877

**Comment by Mistofs** [comment]
Now I understand why the lefties are so suicidal because they can't see progress being made

**Comment by jpryne** [comment]
> Now I understand why the lefties are so suicidal because they can't see progress being made

What does this have to do with political ideology? This is about bleeding edge network protocols being available on the latest operating systems. That is visible progress.
